components/nexus-core/src/main/java/org/sonatype/nexus/proxy/repository/AbstractProxyRepository.java

```diff
@@ -1506,17 +1511,21 @@ protected AbstractStorageItem doRetrieveRemoteItem(ResourceStoreRequest request)
       }
 
       // if we got here, requested item was not retrieved for some reason
+      if (storageCleanupNeeded || request.isRequestRemoteOnly()) {
+        sendContentValidationEvents(request, events, false);
 
-      sendContentValidationEvents(request, events, false);
-
-      try {
-        getLocalStorage().deleteItem(this, request);
-      }
-      catch (ItemNotFoundException e) {
-        // good, we want this item deleted
-      }
-      catch (UnsupportedStorageOperationException e) {
-        log.warn("Unexpected Exception in " + RepositoryStringUtils.getHumanizedNameString(this), e);
+        try {
+          final StorageItem item = getLocalStorage().retrieveItem(this, request);
+          if (!(item instanceof StorageCollectionItem)) {
+            getLocalStorage().deleteItem(this, request);
+          }
+        }
+        catch (ItemNotFoundException e) {
+          // good, we want this item deleted
+        }
+        catch (UnsupportedStorageOperationException e) {
+          log.warn("Unexpected Exception in " + RepositoryStringUtils.getHumanizedNameString(this), e);
```

Comment:
@kellyrob99 Legacy code, not just this class, started on Plexus Logger, and is full of concatenation (and unneeded guards), I don't think we want to spend time changing it...